R/parse-body.R

```diff
@@ -152,36 +196,17 @@ list_parsers <- function() {
   .globals$parsers
 }
 
+#' @describeIn add_parser Select from global parsers and create
+#' a formatted parsers list for programmatic use.
 #' @export
-#' @describeIn add_parser Make named list. Mapping content-type with parser.
-#' @param parser_function A single functions to map to one or more Content-Type.
-#' @param fixed A character vector of fixed string to be matched against a request Content-Type.
-#' @param regex A character vector of [regex] string to be matched against a request Content-Type.
-#' @param shortname A character value to reference a parser by a shortname. (For internal use only)
-#' Content-Type.
-make_parsers <- function(parser_function, fixed = NULL, regex = NULL, shortname = NULL) {
-  if (any(shortname %in% c("fixed", "regex"))) {
-    stop("Shortnames `fixed` and `regex` are reserved for internal use.")
-  }
-  create_list <- function(names) {
-    stats::setNames(
-      replicate(
-        length(names),
-        parser_function),
-      names
-    )
-  }
-  parsers <- list()
-  if (length(shortname) > 0) {
-    parsers$shortname <- create_list(shortname)
-  }
-  if (length(fixed) > 0) {
-    parsers$fixed <- create_list(fixed)
-  }
-  if (length(regex) > 0) {
-    parsers$regex <- create_list(regex)
-  }
-  invisible(parsers)
+select_parsers <- function(alias = character()) {
```

Comment:
Would prefer to make it a required argument

```suggestion
select_parsers <- function(alias) {
```

Reply:
For some reason, github won't let me merge this. Will merge in #584